src/snowflake/snowpark/mock/_stage_registry.py

```diff
@@ -241,3 +448,43 @@ def upload_stream(
             file_name=file_name,
             overwrite=overwrite,
         )
+
+    def get(
+        self,
+        stage_location: str,
+        target_directory: str,
+        options: Dict[str, str] = None,
+    ):
+        if not stage_location.startswith("@"):
+            raise SnowparkSQLException("SQL compilation error")
+        stage_name, stage_prefix = extract_stage_name_and_prefix(stage_location)
+        if stage_name not in self._stage_registry:
+            self.create_or_replace_stage(stage_name)
+
+        return self._stage_registry[stage_name].get_file(
+            stage_location=stage_prefix,
+            target_directory=target_directory,
+            options=options,
+        )
+
+    def read_file(
+        self,
+        stage_location,
+        format: str,
+        schema: List[Attribute],
+        analyzer: "MockAnalyzer",
+        options: Dict[str, str],
+    ):
+        if not stage_location.startswith("@"):
```

Comment:
ditto

Reply:
updated to use more explicit error message